engine/grid.py

```diff
@@ -412,16 +503,14 @@ def get_all_lawnmower_waypoints(self):
         """
         waypoints = []
         node_list = self.nodes
-        rows = node_list.shape[0]
-        cols = node_list.shape[1]
-        for i in range(cols):
-            for j in range(rows):
+        for i in range(self.num_cols):
+            for j in range(self.num_rows):
                 if i % 2 == 0:
                     node = node_list[j, i]
                     self.determine_active_waypoints(node)
                     waypoints.append(node)
                 elif i % 2 == 1:
-                    row_index = rows - (j + 1)
+                    row_index = self.num_rows - (j + 1)
                     node = node_list[row_index, i]
```

Comment:
comment saying this is to add the nodes in reverse order (to not zigzag). I keep forgetting why this is what it is

Reply:
done